driver.go

```diff
@@ -229,70 +256,71 @@ func iso8859(s string) (time.Time, error) {
 	return time.Parse(time.RFC3339, s)
 }
 
-// PreCreateCheck allows for pre-create operations to make sure a driver is ready for creation
+// PreCreateCheck allows for pre-create operations to make sure a driver is
+// ready for creation
 func (d *Driver) PreCreateCheck() error {
-	client, err := d.client()
+	c, err := d.client()
 	if err != nil {
 		return err
 	}
 
-	// TODO find a better "Ping" method
-	_, err = client.CountMachines()
+	_, err = c.Ping(context.Background())
 	if err != nil {
 		return err
 	}
 
-	if _, err := client.GetImage(d.TritonImage); err != nil {
-		// apparently isn't a valid ID, but might be a name like "debian-8" (so let's do a lookup)
+	_, err = c.Images().Get(context.Background(), &compute.GetImageInput{
+		ImageID: d.TritonImage,
+	})
+	if err != nil {
+		// apparently isn't a valid ID, but might be a name like "debian-8" (so
+		// let's do a lookup)
 		// https://github.com/joyent/node-triton/blob/aeed6d91922ea117a42eac0cef4a3df67fbfed2f/lib/tritonapi.js#L368
 		nameVersion := strings.SplitN(d.TritonImage, "@", 2)
 		name, version := nameVersion[0], ""
 		if len(nameVersion) == 2 {
 			version = nameVersion[1]
 		}
-		filter := cloudapi.NewFilter()
-		filter.Add("state", "all")
+
+		listInput := &compute.ListImagesInput{}
+		listInput.State = "all"
```

Comment:
I recall figuring out how to use this filtering API correctly was a pain, so I love seeing that it's improved with a discrete type that has named fields! :+1: :heart: